### PR TITLE
Fixes #7503 user admin create empty google sign in

### LINKED ISSUE #7503: bug: Create users breaks when more than one user has an empty email address for Google Email For Login
This bug was introduced via this PR: https://github.com/openemr/openemr/pull/7472

The default setting for the insert is an empty string instead of NULL.  As the column as a unique key on it from the 6.0.0 release it requires all values to be unique.  If you create two users and leave the Google Email for Login blank, the first user will create successfully populating the unique key and the second user will fail because there already exists a unique key.  

Note it was difficult to track this down because the sql.inc.php file is not getting the $last_mysql_error.  The error log even gives a warning here:
```
PHP Warning:  Undefined global variable $last_mysql_error in /var/www/localhost/htdocs/openemr/library/sql.inc.php on line 818, referer: https://localhost:9300/interface/usergroup/usergroup_admin_add.php
```

### PR DESCRIPTION
**Pull request by adunsulag**
Fixes #7503 by allowing the multiple users to have an empty google sign in email address. If the address is empty it sets the value to null which doesn't trigger the unique constraint.

### CODE REVIEW

**Comment by bradymiller** [comment]
Weird that this is failing ci e2e test of adding user. Code looks fine. Is this testing well?

File: interface/usergroup/usergroup_admin.php [hunk 1]
```diff
@@ -351,8 +359,8 @@
             "', mname = '"         . add_escape_custom(trim((isset($_POST['mname']) ? $_POST['mname'] : ''))) .
             "', lname = '"         . add_escape_custom(trim((isset($_POST['lname']) ? $_POST['lname'] : ''))) .
             "', suffix = '"         . add_escape_custom(trim((isset($_POST['suffix']) ? $_POST['suffix'] : ''))) .
-            "', google_signin_email = '" . add_escape_custom(trim((isset($_POST['google_signin_email']) ? $_POST['google_signin_email'] : ''))) .
-            "', valedictory = '"         . add_escape_custom(trim((isset($_POST['valedictory']) ? $_POST['valedictory'] : ''))) .
+            "', google_signin_email = " . add_escape_custom($googleSigninEmail) .
+            ", valedictory = '"         . add_escape_custom(trim((isset($_POST['valedictory']) ? $_POST['valedictory'] : ''))) .
```
Comment (stephenwaite) [inline interface/usergroup/usergroup_admin.php:363]: looks like something with a missing `'`

**Comment by bradymiller** [comment]
Another option is to convert the statement to use binding (that deals with null without needing to worry about single quotes etc) if keep getting this weird ci fail despite testing well.

**Comment by adunsulag** [comment]
@bradymiller I think I'll do that, I hate using the string concatenation and would prefer actual prepared statements here.

File: interface/usergroup/usergroup_admin.php (hunk 1 above)
Comment (adunsulag) [reply interface/usergroup/usergroup_admin.php:363]: I add the quotes higher up, but going to change it to binding, seems better that way overall.

File: interface/usergroup/usergroup_admin.php [hunk 2]
```diff
@@ -343,6 +343,14 @@
         }
 
         if ($doit == true) {
+            // google_signin_email has unique key constraint, needs to be handled differently
+            if (isset($_POST["google_signin_email"])) {
+                if (empty($_POST["google_signin_email"])) {
+                    $googleSigninEmail = null;
```
Comment (stephenwaite) [inline interface/usergroup/usergroup_admin.php:349]: or escape the single quotes here?

**Comment by adunsulag** [comment]
@bradymiller @stephenwaite ended up backing out the parameter binding as I didn't want to mess with AuthUtils::updatePassword which is where the user insertion is going.  Its been heavily vetted and I worried about breaking something even though that function is doing a lot of heavy lifting.

**Review by bradymiller — APPROVED** [review]

**Pull request merged**